app/src/main/java/com/teamagam/gimelgimel/app/settings/dialogs/SetUsernameAlertDialogBuilder.java

```diff
@@ -0,0 +1,108 @@
+package com.teamagam.gimelgimel.app.settings.dialogs;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.text.Editable;
+import android.text.InputFilter;
+import android.text.InputType;
+import android.text.Spanned;
+import android.text.TextWatcher;
+import android.widget.EditText;
+
+import com.teamagam.gimelgimel.R;
+import com.teamagam.gimelgimel.app.common.utils.UsernameGenerator;
+import com.teamagam.gimelgimel.app.settings.fragments.GeneralPreferenceFragment;
+
+
+public class SetUsernameAlertDialogBuilder {
+
+    public static final InputFilter EMOJI_FILTER = new InputFilter() {
+
+        @Override
+        public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
+            for (int index = start; index < end; index++) {
+                int type = Character.getType(source.charAt(index));
+                if (type == Character.SURROGATE) {
+                    return "";
```

Comment:
"" and null can be extracted into static final fields with names FILTER and NO_FILTER to help the reader understand. 
I managed to understand their meaning only after I read the filter's method description